### PR TITLE
Remove persistence bean

### PR DESCRIPTION
**Pull request by alexshtin**
## What changed?

Remove persistence bean.

## Why?

Legacy unused code.

## How did you test it?

Existing tests.

## Potential risks

No risks.

## Documentation

No.

## Is hotfix candidate?

No.

### CODE REVIEW

**Comment by dnr** [comment]
Does this work? Why did Michael's attempt in #5000 have to add/move a bunch of new code but this doesn't?

**Comment by tdeebswihart** [comment]
> Does this work? Why did Michael's attempt in #5000 have to add/move a bunch of new code but this doesn't?

It doesn't, as it seems to have broken our PostgreSQL backend entirely

**Comment by alexshtin** [comment]
I guess it is one `Close()` call

**Comment by alexshtin** [comment]
Yeah, it was `factory.Close()` call.

File: common/persistence/client/fx.go [hunk 1]
```diff
@@ -152,3 +167,77 @@ func HealthSignalAggregatorProvider(
...
 	return persistence.NoopHealthSignalAggregator
 }
+
+func ClusterMetadataManagerProvider(factory Factory) (persistence.ClusterMetadataManager, error) {
+	return factory.NewClusterMetadataManager()
+}
+func ClusterMetadataManagerLifetimeHooks(
+	lc fx.Lifecycle,
+	manager persistence.ClusterMetadataManager,
+) {
```
Comment (dnr) [inline common/persistence/client/fx.go:177]: these would look a little nicer on one line (rename `manager` to `m` if you want to stay ≤ 100 columns)

**Review by dnr — APPROVED** [review]

**Pull request merged**